opta/commands/generate_terraform.py

```diff
@@ -0,0 +1,304 @@
+        REGISTRY[layer.cloud]["backend"] = {"local": {"path": backend_dir}}
+        # generate the main.tf.json
+        try:
+            execution_plan = list(gen(layer))
+        finally:
+            REGISTRY[layer.cloud]["backend"] = original_backend
+
+        # break down json file in multiple files
+        main_tf_json = json.load(open(TF_FILE_PATH))
+        for key in ["provider", "data", "output", "terraform"]:
+            # extract the relevant json
+            main_tf_json, extracted_json = dicts.extract(main_tf_json, key)
+
+            # if there was already a file for it, merge it
+            # ex: combine all the terraform "output" variables
+            prev_tf_file = os.path.join(output_dir, f"{key}.tf.json")
+            if os.path.exists(prev_tf_file):
+                logger.debug(
+                    f"Found existing terraform file: {prev_tf_file}, merging it with new values"
+                )
+                prev_json = json.load(open(prev_tf_file))
+                extracted_json = dicts.merge(extracted_json, prev_json)
+
+            # save it as it's own file
+            _write_json(extracted_json, os.path.join(tmp_dir, f"{key}.tf.json"))
+
+        # extract modules tf.json in their own files
+        main_tf_json, modules_json = dicts.extract(main_tf_json, "module")
+        for name, value in modules_json["module"].items():
+            _write_json(
+                {"module": {name: value}}, os.path.join(tmp_dir, f"module-{name}.tf.json")
+            )
+
+        # update the main file without the extracted sections
+        if main_tf_json:
+            # only write file there is anything remaining
+            _write_json(
+                main_tf_json, os.path.join(tmp_dir, f"{tmp_dir}/{layer.name}.tf.json")
+            )
+
+        # generate the readme
+        opta_cmd = f"opta {ctx.info_name} {str_options(ctx)}"
+        readme_file = _generate_readme(
+            layer, execution_plan, tmp_dir, readme_format, opta_cmd
+        )
+
+        # if everything was successfull, copy tmp dir to target dir
+        if os.path.exists(output_dir):
+            if replace:
+                logger.info(
+                    f"Output directory {directory} already exists and --replace flag is on, deleting it"
+                )
+                if not auto_approve:
+                    click.confirm(
+                        f"The existing directory will be deleted: {output_dir}.\n Do you approve?",
+                        abort=True,
+                    )
+                _clean_folder(output_dir)
+            else:
+                logger.info(
+                    f"Output directory {directory} already exists, adding new files to it"
+                )
+
+        logger.debug(f"Copy {tmp_dir} to {output_dir}")
+        shutil.copytree(tmp_dir, output_dir, dirs_exist_ok=True)
+        logger.info("Terraform files generated successfully.")
+        if readme_file:
```

Comment:
really good point.
Here is what I've done:
- a module can be marked as `is_exportable: false` by default it would be true
- if generate terraform includes a unsupported module:
  - command output will be: Terraform files partially generated, a few modules are not supported:
  - in the generated readme, the  unsupported modules will be listed
  - if we want to include special documentation as well, we can define a file called `export.md` in the module directory, this file will be added to the generated readme. I added some examples in this PR with datadog and aws-dns

as we ramp up the testing, we will mark deal with the unsupported module:
- either mark them as `is_exportable: false`
- or refactor them to be exportable